unstructured_ingest/embed/huggingface.py

```diff
@@ -54,7 +54,9 @@ def _embed_query(self, query: str) -> list[float]:
         return self._embed_documents(texts=[query])[0]
 
     def _embed_documents(self, texts: list[str]) -> list[list[float]]:
-        client = self.config.get_client()
+        if not hasattr(self, "_cached_client"):
```

Comment:
i do agree that w should not create a new client everytime, it is too expensive. should see what is the right way to cache if thats the right way. have to see the usage pattern to determine the answer

Reply:
ok the client should be moved to the `embed_documents` function and the client should be passed in as a variable to this one